src/controllers/userController.ts

```diff
@@ -62,4 +63,75 @@ const login = async (
   res.status(StatusCodes.OK).json({ user: { name: user.name }, token });
 };
 
-export { register, login };
+const getUserProfile = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+
+    const { email } = req.body;
```

Comment:
const email = req.query.email as string; With GET requests, parameters are passed in the URL, so it’s commonly done with query parameters and rewrite Swagger after this

Reply:
fixed. new reques should look like htttp://localhost:8000/api/v1/auth/profile?email=ivan@ivan.c1om